File: Katas.FSharp/SpeedControl.fs
module SpeedControl

let speed distance period = int ((3600. * distance) / (float period))

let gps period distances =
    match distances with
    | [] -> 1
    | [_] -> 1
    | xs ->
        xs
        |> Seq.pairwise
        |> Seq.map (fun (a,b) -> speed (b-a) period)
        |> Seq.max

let rec gpsOther period distances =
    match distances with
    | [] -> 1
    | _ :: [] -> 1
    | a :: b :: xs -> max (speed (b-a) period) (gpsOther period (b :: xs))

File: Codewars/SpeedControl.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Katas.CSharp
{
    public class SpeedControl
    {
        public static int Gps(int period, IEnumerable<double> distances)
            => distances.Aggregate(new {value = .0, speed = 1},
                (speedAndValue, nextValue) => new
                {
                    value = nextValue,
                    speed = Math.Max((int) (3600 * (nextValue - speedAndValue.value) / period), speedAndValue.speed)
                }, speedAndValue => speedAndValue.speed);
    }
}